osdep/win32/pthread.c

```diff
@@ -77,6 +76,23 @@ int pthread_mutex_unlock(pthread_mutex_t *mutex)
     return 0;
 }
 
+int clock_gettime(clockid_t clockid, struct timespec *tp)
+{
+    if (clockid != CLOCK_REALTIME) {
+        errno = EINVAL;
+        return -1;
+    }
+    union {
+        FILETIME ft;
+        ULARGE_INTEGER i;
+    } r;
+    GetSystemTimePreciseAsFileTime(&r.ft);
```

Comment:
> Anyway I don't think doing the extra work to support Windows 7 here is justified.

I'm still not sure I agree. It is one check. And will save us a lot of headache explaining to people with Windows 7 not working anymore.

``` C
static __typeof__(&GetSystemTimePreciseAsFileTime) pGetSystemTimePreciseAsFileTime = NULL;
static void init_gstpast(void)
{
    HMODULE k = GetModuleHandleW(L"kernel32.dll");
    if (!k) return;
    pGetSystemTimePreciseAsFileTime = GetProcAddress(k, "pGetSystemTimePreciseAsFileTime");
}

int clock_gettime(clockid_t clockid, struct timespec *tp) {
    ...
    static pthread_once_t once_init_gstpast = PTHREAD_ONCE_INIT;
    pthread_once(&once_init_gstpast , init_gstpast);
    ...
    if (!pGetSystemTimePreciseAsFileTime ) {
        timespec_get(&ts, TIME_UTC);
        ...
    }
    ...
}
```